### PR TITLE
initial tests for homeController

### PR DESCRIPTION
**Pull request by AntsiferauGodel**
Initial tests for home Controller

### CODE REVIEW

File: Lab08.MVC/Lab08.MVC.Web.Tests/HomeControllerTests.cs [hunk 1]
```diff
@@ -113,14 +185,221 @@ public void AdvertisementDetails_CorrectValues_ResultNotNull()
...
+        public void AdvertisementsForCurrentUser_CorrectValues_ResultNotNull()
+        {
+            var result = homeController.AdvertisementsForCurrentUser("UserId");
+            Assert.NotNull(result);
         }
 
         [Test]
-        public void AdvertisementDetails_CorrectValues_ReturnRedirectToRouteResult()
+        public void AdvertisementsForCurrentUser_CorrectValues_InvokesAdvertisementServiceGetAdvertisementsTotalPagesNumberForCurrentUser()
         {
```
Comment (HannaKeryna) [inline Lab08.MVC/Lab08.MVC.Web.Tests/HomeControllerTests.cs]: 118 symbols
I suggest to use shorter names

File: Lab08.MVC/Lab08.MVC.Web.Tests/HomeControllerTests.cs [hunk 2]
```diff
@@ -113,14 +185,221 @@ public void AdvertisementDetails_CorrectValues_ResultNotNull()
...
+        {
+            var result = homeController.AdvertisementRemove(1);
             Assert.IsInstanceOf(typeof(RedirectToRouteResult), result);
         }
+
+        [Test]
+        public void AdvertisementRemove_CorrectValues_InvokesAdvertisementServiceRemoveAdvertisementById()
+        {
+            homeController.AdvertisementRemove(1);
+            advertisementServiceMock.Verify(m => m.RemoveAdvertisementById(It.IsAny<int>()));
```
Comment (HannaKeryna) [inline Lab08.MVC/Lab08.MVC.Web.Tests/HomeControllerTests.cs]: Where at least one assert?

File: Lab08.MVC/Lab08.MVC.Web.Tests/HomeControllerTests.cs [hunk 3]
```diff
@@ -113,14 +185,221 @@ public void AdvertisementDetails_CorrectValues_ResultNotNull()
...
+            var result = homeController.AdvertisementSearch("SearchWord");
+            Assert.NotNull(result);
+        }
+
+        [Test]
+        public void AdvertisementSearch_CorrectValues_InvokesAdvertisementServiceGetAdvertisementsTotalPagesNumberByPartialHeader()
+        {
+            homeController.AdvertisementSearch("SearchWord");
+            advertisementServiceMock
+                .Verify(m => m.GetAdvertisementsTotalPagesNumberByPartialHeader(It.IsAny<string>(), It.IsAny<int>()));
```
Comment (HannaKeryna) [inline Lab08.MVC/Lab08.MVC.Web.Tests/HomeControllerTests.cs]: And in this test

File: Lab08.MVC/Lab08.MVC.Web/Controllers/HomeController.cs [hunk 4]
```diff
@@ -142,7 +146,10 @@ public ActionResult AdvertisementEdit(int advertisementId)
         public ActionResult AdvertisementEdit(AdvertisementView advertisement, HttpPostedFileBase[] upload)
         {
             #region securityCheck
-            CheckOwnerShip(advertisement.OwnerId);
+            if (IsInvalidOwner(advertisement.OwnerId))
+            {
+                return View("NotFound", "ErrorHandler");
+            }
             #endregion
```
Comment (HannaKeryna) [inline Lab08.MVC/Lab08.MVC.Web/Controllers/HomeController.cs]: Please look at this method at if/elase. You can use:
advertisementEntity.Files = advertisement.Pictures != null 
    ? advertisementEntity.Files.Concat(files).ToList() 
    : files;

File: Lab08.MVC/Lab08.MVC.Web.Tests/HomeControllerTests.cs [hunk 5]
```diff
@@ -1,63 +1,128 @@
...
+            homeController.ModelState.Clear();
         }
 
         [Test]
-        public void Index_Void_ReturnActionResult()
+        public void Index_Null_NotNull()
         {
             var result = homeController.Index();
-            Assert.IsInstanceOf(typeof(ActionResult), result);
+            Assert.NotNull(result);
```
[the remark this answers is not in the document]
Comment (AntsiferauGodel) [reply Lab08.MVC/Lab08.MVC.Web.Tests/HomeControllerTests.cs]: done

File: Lab08.MVC/Lab08.MVC.Web.Tests/HomeControllerTests.cs (hunk 1 above)
Comment (AntsiferauGodel) [reply Lab08.MVC/Lab08.MVC.Web.Tests/HomeControllerTests.cs]: done

File: Lab08.MVC/Lab08.MVC.Web/Controllers/HomeController.cs (hunk 4 above)
Comment (AntsiferauGodel) [reply Lab08.MVC/Lab08.MVC.Web/Controllers/HomeController.cs]: done

File: Lab08.MVC/Lab08.MVC.Web.Tests/HomeControllerTests.cs [hunk 6]
```diff
@@ -1,63 +1,128 @@
...
+            };
+            invalidAdvertisementViewModel = new AdvertisementView
+            {
+                Header = null,
+                Description = null,
+                Price = 0
+            };
+            fakeAdvertisements = new List<Advertisement> {
+                fakeAdvertisement,
+                fakeAdvertisement
```
[the remark this answers is not in the document]
Comment (AntsiferauGodel) [reply Lab08.MVC/Lab08.MVC.Web.Tests/HomeControllerTests.cs]: done

File: Lab08.MVC/Lab08.MVC.Web.Tests/HomeControllerTests.cs (hunk 2 above)
Comment (AntsiferauGodel) [reply Lab08.MVC/Lab08.MVC.Web.Tests/HomeControllerTests.cs]: discussed in skype

File: Lab08.MVC/Lab08.MVC.Web.Tests/HomeControllerTests.cs (hunk 3 above)
Comment (AntsiferauGodel) [reply Lab08.MVC/Lab08.MVC.Web.Tests/HomeControllerTests.cs]: discussed in skype

**Pull request merged**